modules/apps/site/site-cms-site-initializer/src/main/java/com/liferay/site/cms/site/initializer/internal/frontend/data/set/filter/AuthorSelectionFDSFilter.java

```diff
@@ -29,39 +45,67 @@
 )
 public class AuthorSelectionFDSFilter extends BaseSelectionFDSFilter {
 
-	@Override
-	public String getAPIURL() {
-		return "/o/headless-admin-user/v1.0/user-accounts?sort=name:asc";
-	}
-
 	@Override
 	public String getEntityFieldType() {
 		return FDSEntityFieldTypes.INTEGER;
 	}
 
 	@Override
 	public String getId() {
-		return "creatorId";
+		return "userId";
 	}
 
 	@Override
-	public String getItemKey() {
-		return "id";
+	public String getLabel() {
+		return "author";
 	}
 
 	@Override
-	public String getItemLabel() {
-		return "name";
-	}
+	public List<SelectionFDSFilterItem> getSelectionFDSFilterItems(
+		Locale locale) {
 
-	@Override
-	public String getLabel() {
-		return "author";
+		List<User> users = _userLocalService.search(
```

Comment:
If you use the local service, you should do the same kind of role check we do in the UserAccountResource.